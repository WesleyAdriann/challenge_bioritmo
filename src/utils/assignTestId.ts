type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
)  => ({
    'data-testid': `${componentType}_${componentId}`,
  })


export const mockTestID = (
    componentType: componentType,
    componentId: string
  ) => assignTestId(componentType, componentId)
