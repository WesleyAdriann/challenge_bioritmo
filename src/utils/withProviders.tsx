
import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../style'

export const StorybookProviders: React.FC = ({
  children,
}) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
)
