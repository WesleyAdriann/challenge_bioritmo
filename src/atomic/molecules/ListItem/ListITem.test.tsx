import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'

import { mockTestID, StorybookProviders as wrapper } from '../../../utils'

import { ListItem, IListItem } from './index'

describe('Molecules/ListItem', () => {
  const props: IListItem = {
    testID: 'ListItem',
    text: 'Text Test'
  }

  const renderComponent = () => render(<ListItem {...props} />, { wrapper })

  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  const el_container = mockTestID('div', props.testID!)
  const el_text = mockTestID('text', `${props.testID}_text`)
  const el_right_text = mockTestID('text', `${props.testID}_right_text`)

  it(`should render component #${el_container}`, async () => {
    const sut = await component.findByTestId(el_container)
    expect(sut).toBeTruthy()
  })

  it('should render text', async () => {
    component.rerender(<ListItem {...props} />)
    const sut = await component.findByTestId(el_text)
    expect(sut).toBeTruthy()
    expect(sut.textContent).toBe(props.text)
  })

  it('should render right text on pass prop', async () => {
    const rightText = 'Right Text test'
    component.rerender(<ListItem {...props} rightText={rightText} />)
    const sut = await component.findByTestId(el_right_text)
    expect(sut).toBeTruthy()
    expect(sut.textContent).toBe(rightText)
  })

  it('should call onClick prop on click if pass onClick prop', async () => {
    const onClick = jest.fn()
    component.rerender(<ListItem {...props} onClick={onClick} />)
    const sut = await component.findByTestId(el_container)
    fireEvent.click(sut)
    expect(onClick).toBeCalled()
  })

  it('should render snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot()
  })
})
