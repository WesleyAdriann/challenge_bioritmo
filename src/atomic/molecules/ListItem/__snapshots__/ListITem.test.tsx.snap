// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/ListItem should render snapshot 1`] = `
.c2 {
  color: #000;
  font-weight: 400;
  font-size: 14px;
  text-align: left;
  text-transform: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: block;
  line-height: normal;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #F3F3F3;
  cursor: initial;
  margin-bottom: 0px;
  padding: 8px 0;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  class="c0"
  data-testid="div_ListItem"
>
  <div
    class="c1"
  >
    <p
      class="c2"
      data-testid="text_ListItem_text"
    >
      Text Test
    </p>
  </div>
  <p
    class="c2"
    data-testid="text_ListItem_right_text"
  />
</div>
`;
